video/image_writer.c

```diff
@@ -163,9 +164,10 @@ static void prepare_avframe(AVFrame *pic, AVCodecContext *avctx,
     );
 }
 
-static bool write_lavc(struct image_writer_ctx *ctx, mp_image_t *image, const char *filename)
+static bool write_lavc(struct image_writer_ctx *ctx, mp_image_t *image,
+                       const char *filename, bool overwrite)
 {
-    FILE *fp = fopen(filename, "wb");
+    FILE *fp = fopen(filename, overwrite ? "wb" : "wbx");
     if (!fp) {
         MP_ERR(ctx, "Error opening '%s' for writing!\n", filename);
```

Comment:
nit, but now that in other place you print that file already exist, check for `EEXIST` and do here too?